rust/scx_rustland_core/assets/bpf/main.bpf.c

```diff
@@ -462,19 +466,23 @@ static s32 pick_idle_cpu(struct task_struct *p, s32 prev_cpu)
 		 * Search for any full-idle CPU in the task domain that shares
 		 * the same L2 cache.
 		 */
-		cpu = bpf_cpumask_any_and_distribute(cast_mask(l2_mask), idle_smtmask);
-		if (bpf_cpumask_test_cpu(cpu, online_cpumask) &&
-		    scx_bpf_test_and_clear_cpu_idle(cpu))
-			goto out_put_cpumask;
+		if (l2_mask) {
+			cpu = bpf_cpumask_any_and_distribute(cast_mask(l2_mask), idle_smtmask);
+			if (bpf_cpumask_test_cpu(cpu, online_cpumask) &&
+				scx_bpf_test_and_clear_cpu_idle(cpu))
```

Comment:
Can you align this to bpf_cpumask_test_cpu() since it's part of the condition? Otherwise it looks like the first instruction of the if block.